posts/integrating-civic-into-a-static-serverless-website-part-1-of-2.md

```diff
@@ -0,0 +1,231 @@
+    console.info(`Civic anonymous identity token: ${anonymousIdentityToken}`);
+    console.info(`Decoded token: ${JSON.stringify(JSON.parse(atob(anonymousIdentityToken.split('.')[1])), null, 4)}`);
+
+    // pass the token to our backend for processing
+    getColorIdentity(anonymousIdentityToken);
+});
+```
+
+The decoding of the token above is just for debugging. You can see how we can't yet access the identity information inside the token. Civic requires the token to be verified on the backend before sharing the user's data. It just so happens that in this use case, we eventually pass back the user ID but we don't have to.
+
+![missing pic: Civic JWT Token](https://raw.githubusercontent.com/tylerjohnhaden/blog-usa/master/images/2019/02/civic_jwt_token.png)
+
+We can get a little bit of insight into the token by decoding, like that it expires after 30 minutes. Here, `codeToken` will eventually be converted into our user id. For more information on JWT in general, checkout the [jwt.io docs](https://jwt.io/introduction/).
+
+Next, we send this token to our backend. There is nothing wrong with passing the token back in the body, but the standard way is using the `Authorization` header. This also makes it simpler for us to use AWS Lambda Authorizers (although they can use custom locations for the token). 
+
+This can be accomplished with a simple GET request with authorization credentials specified.
+
+```javascript
+function getColorIdentity(token) {
+    fetch(apiColorIdentificationEndpoint, {
+        withCredentials: true,
+        credentials: 'include',
+        headers: {
+            'Authorization': token,
+            'Content-Type': 'application/json'
+        }
+
+    }).then(response => response.json()).then(identities => {
+        console.info(`Received anonymous identity information from API: ${JSON.stringify(identities, null, 4)}`);
+
+        // convert integer into hex color format
+        const hexValue = '#' + identities.yourColorIdentity.toString(16).padStart(6, '0');
+
+        // display color
+        document.body.style.backgroundColor = hexValue;
+
+        setTimeout(() => {
+            alert(`Your personal identity is ${identities.yourUserIdentity}!\n\n` +
+                  `Your color identity is ${hexValue}!`);
+        }, 20);
+
+    // or display error
+    }).catch(err => console.error(err));
+}
+```
+
+Our backend will turn the Civic JWT token into an anonymous user identity, and color identity. The user ID is just for debugging, and our use case uses the color identity to set the background color of the web page.
+
+The Javascript [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) and most HTTP request APIs abstract away the CORS negotiation. Since our backend is at a different origin (even sub-domains count as cross-origin), there is a request to `OPTIONS` that will return various important headers agreeing with our credentials and cross-origin request, more on this in the next section.
```

Comment:
```suggestion
The Javascript [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) and most HTTP request APIs abstract away the CORS negotiation. Since our backend is at a different origin (even sub-domains count as cross-origin), there is a request to `OPTIONS` that will return various important headers agreeing with our credentials and cross-origin request. More on this in the next section.
```